sdk/mobile-center-analytics/src/main/java/com/microsoft/azure/mobile/analytics/Analytics.java

```diff
@@ -385,9 +399,65 @@ private synchronized void setInstanceAutoPageTrackingEnabled(boolean autoPageTra
     /**
      * Implements {@link #setListener(AnalyticsListener)}.
      */
-    private synchronized void setInstanceListener(AnalyticsListener listener)
-    {
+    private synchronized void setInstanceListener(AnalyticsListener listener) {
         mAnalyticsListener = listener;
     }
 
+    /**
+     * Validates name.
+     *
+     * @param name log name to validate.
+     * @return <code>true<code/> if validation succeeds, otherwise <code>false</code>.
+     */
+    private static boolean validateName(String name) {
+        final int maxNameLength = 256;
+        if (name == null || name.isEmpty()) {
+            MobileCenterLog.error(Analytics.LOG_TAG, "Name cannot be null or empty.");
+            return false;
+        }
+        if (name.length() > maxNameLength) {
+            MobileCenterLog.error(Analytics.LOG_TAG, "Name length cannot be longer then " + maxNameLength + " characters.");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Validates properties.
+     *
+     * @param properties properties collection to validate.
+     * @return valid properties collection with maximum size of 5
+     */
+    private static Map<String, String> validateProperties(Map<String, String> properties) {
+        if (properties == null)
+            return null;
+        final int maxPropertiesCount = 5;
+        final int maxPropertyItemLength = 64;
+        Map<String, String> result = new HashMap<>();
+        for (Map.Entry<String, String> property : properties.entrySet()) {
+            if (property.getKey() == null || property.getKey().isEmpty()) {
+                MobileCenterLog.error(Analytics.LOG_TAG, "Property key cannot be null or empty. Property will be skipped.");
+                continue;
+            }
+            if (property.getKey().length() > maxPropertyItemLength) {
+                MobileCenterLog.error(Analytics.LOG_TAG, "Property key length cannot be longer than " + maxPropertyItemLength + " characters. Property " + property.getKey() + " will be skipped.");
+                continue;
+            }
+            if (property.getValue() == null) {
+                MobileCenterLog.error(Analytics.LOG_TAG, "Property value cannot be null. Property " + property.getKey() + " will be skipped.");
+                continue;
+            }
+            if (property.getValue().length() > maxPropertyItemLength) {
+                MobileCenterLog.error(Analytics.LOG_TAG, "Property value length cannot be longer than " + maxPropertyItemLength + " characters. Property " + property.getKey() + " will be skipped.");
```

Comment:
`Event '{name}' : property '{key}' : property value cannot be longer than 64 characters. Property '{key}' will be skipped.`